src/xcomponentAPI.js

```diff
@@ -0,0 +1,26 @@
+
+define(["communication/connection", "parser", "configuration/xcomponentConfiguration", "communication/xcomponentPublisher", "rx"], function (Connection, Parser, XComponentConfiguration, Publisher, Rx) {
+    "use strict";
+
+    var Factory = function () {
+        this.createXComponentApi = function (serverUrl, callback, callbackError) {
```

Comment:
Make code simpler.
Just create a factory method.
move createXComponentApi function as a static method of XComponentAPI
Better naming in this case createApi